--- twinkle-0.8/src/gui/userprofileform.ui.orig	2006-06-24 22:56:53.000000000 +0200
+++ twinkle-0.8/src/gui/userprofileform.ui	2006-07-04 01:12:02.446645500 +0200
@@ -3955,6 +3955,7 @@
     <tabstop>categoryListBox</tabstop>
 </tabstops>
 <includes>
+    <include location="local" impldecl="in declaration">map</include>
     <include location="local" impldecl="in declaration">user.h</include>
     <include location="local" impldecl="in declaration">qvaluelist.h</include>
     <include location="local" impldecl="in implementation">qlistbox.h</include>
@@ -3978,7 +3979,7 @@
     <include location="local" impldecl="in implementation">userprofileform.ui.h</include>
 </includes>
 <variables>
-    <variable access="private">map&lt;t_user *, int&gt; map_last_cat;</variable>
+    <variable access="private">std::map&lt;t_user *, int&gt; map_last_cat;</variable>
     <variable access="private">t_user *current_profile;</variable>
     <variable access="private">int current_profile_idx;</variable>
     <variable access="private">list&lt;t_user *&gt; profile_list;</variable>
