diff -urN twinkle-0.7.1.org/src/gui/gui.cpp twinkle-0.7.1/src/gui/gui.cpp
--- twinkle-0.7.1.org/src/gui/gui.cpp	2006-05-05 22:39:08.000000000 +0200
+++ twinkle-0.7.1/src/gui/gui.cpp	2006-06-12 11:56:45.000000000 +0200
@@ -734,8 +734,19 @@
 	
 	// Incoming call for to-header
 	mainWindow->displayHeader();
+
+       // From
+        QString fromParty = format_sip_address(user_config,
+                                r->hdr_from.display, r->hdr_from.uri).c_str();
+        if (r->hdr_organization.is_populated()) {
+                fromParty.append(", ").append(r->hdr_organization.name.c_str());
+        }
+
+	
 	s = "Line ";
-	s.append(QString().setNum(line + 1)).append(":\tincoming call for ");
+	s.append(QString().setNum(line + 1)).append(":\tincoming call from ");
+	s.append(fromParty);
+	s.append(" for ");
 	s.append(format_sip_address(user_config, r->hdr_to.display, r->hdr_to.uri).c_str());
 	mainWindow->display(s);
 	
